packages/redux/src/bags/selectors.ts

```diff
@@ -107,13 +107,17 @@ export const getBagId = (state: StoreState): State['id'] => getId(state.bag);
 
 export const getBagItem = createSelector(
   [
-    (state: StoreState, bagItemId: BagItem['id']): BagItemEntity =>
-      getEntityById(state, 'bagItems', bagItemId),
+    (state: StoreState, bagItemId: BagItem['id']) =>
+      getEntityById(state, 'bagItems', bagItemId) as BagItemEntity,
     (
       state: StoreState,
       bagItemId: BagItem['id'],
     ): ProductEntity | undefined => {
-      const bagItem = getEntityById(state, 'bagItems', bagItemId);
+      const bagItem = getEntityById(
+        state,
+        'bagItems',
+        bagItemId,
+      ) as BagItemEntity;
```

Comment:
On "packages/redux/src/wishlists/selectors/wishlistsSets.ts" you removed the cast to `WishlistSetEntity` in favor of typing the const like `const wishlistSet: WishlistSetEntity | undefined = ...`. Can't we do the same here?

Reply:
I tried doing the same here, but doing that i get an error in this line `return getProduct(state, bagItem?.product);` because bagItem can be undefined and the `getProduct` always expects a number.

Since it's not my area, I wasn't confortable changing this, so I used that as a workaround. But I agree with you, I don't like this approach too. I was thinking doing something like this:

`return getProduct(state, bagItem?.product || 0);`

Which fixes the error, but I'm afraid that it breaks something. What do you think?

Reply:
I also think it will break something ahah. Could you please add an "@TODO" on that selector so my team knows there is work to do there? 🙏 Thanks!

Reply:
Yes no problem :)